modules/serverless-rn-1-17-0.adoc

```diff
@@ -0,0 +1,60 @@
++
+[source,yaml]
+----
+...
+spec:
+  config:
+    network:
+      defaultExternalScheme: "http"
+...
+----
+
+* mTLS functionality is now Generally Available (GA).
+
+* TypeScript templates are now available when you create a function using `kn func`.
+
+* Changes to API versions in Knative Eventing 0.23.0:
+
+** Version `v1beta1` has been removed from the `eventing.knative.dev`, `messaging.knative.dev` and `flows.knative.dev` APIs. You must ensure that any existing or new resources in your deployment that are created by using these APIs use version `v1`.
+
+** The `v1alpha1` version of the `KafkaChannel` API, which was deprecated in {ServerlessProductName} version 1.14.0, has been removed. If the `ChannelTemplateSpec` of your config maps contain references to this older version, you must update this part of the spec to use the correct API version.
+
+** The deprecated `v1alpha1`, `v1alpha2`, and `v1beta1` versions of the `APIServerSource` and `SinkBinding` APIs have been removed. You must update any existing resources to use the correct API versions.
+
+** The deprecated `v1alpha2` and `v1beta1` versions of the `ContainerSource` and `PingSource` APIs have been removed. You must update any existing resources to use the correct API versions.
+
+** Version `v1beta2` of the `PingSource` API has been deprecated. References to older versions of this API should be updated to `v1` to prepare your deployment for removal of the deprecated version.
+
+[id="fixed-issues-1-17-0_{context}"]
+== Fixed issues
+
+[id="known-issues-1-17-0_{context}"]
+== Known issues
+
+* If you try to use an older version of the Knative `kn` CLI with a newer {ServerlessProductName} release, the API is not found and an error occurs.
++
+For example, if you use the 1.16.0 release of the `kn` CLI, which uses version 0.22.0, with the 1.17.0 {ServerlessProductName} release, which uses the 0.23.0 versions of the Knative Serving and Knative Eventing APIs, the CLI does not work because it continues to look for the outdated 0.22.0 API versions.
++
+Ensure that you are using the latest `kn` CLI version for your {ServerlessProductName} release to avoid issues.
+
+* If you create a new subscription for a Kafka channel, there might be a delay in the Kafka channel data plane becoming ready to dispatch messages after the newly created subscription reports a ready status. As a result, messages that are sent during the time which the data plane is not reporting a ready status might not be delivered to the subscriber.
++
+The following steps are recommended to mitigate this issue:
+** Schedule a maintenance window for updates, during which the Kafka channel is not expected to receive events.
+** If you are creating subscriptions manually, send a few test events to ensure that events are not being lost. After you have confirmed that test events are not being lost, it is safe to continue using your subscriptions as usual.
```

Comment:
> We can't advise sending test events. That maybe done on testing or staging environment, but any respectable company will not allow such data forging in production!

It really depends heavily on the use case and the technique. Some use cases allow for such manual events ingestion in a production system cause events are ephemeral events. It depends heavily. Definitely we don't recommend them in cases that would invalidate the customer use case. So maybe we can be explicit in the recommendation and add "when possible" or some other wording that make this recommendation conditional. 

> Instead, we should instruct what to check on the environment - I presume checking underlying Kafka topic for committed offsets should be enough of a test that KafkaChannel/KafkaSource is now fully ready.

that is some extra recommendation, there are a lot of specific options we can add so maybe we can generalize them to 'verify events receipt either via safe validation events (heartbeat, ephemeral events, test events) if your use case allows it or via checking Kafka topic for committed offsets'

Reply:
I agree w/ Ahmed here. It all really depends...

Reply:
I updated the suggestion here https://github.com/openshift/openshift-docs/pull/35500#discussion_r707636599

Reply:
It doesn't. We should provide solution viable for even the most regulated industries, the worst case. Such solution could be used also in less regulated ones.

Reply:
What I meant by "it depends" is whether or not having that advise would lead to "data forging" . And I guess "it depends" means it might be problematic for some use cases, and not be for other use case. So "it does depend".
 
> any respectable company will not allow such data forging in production!

* "Respectable company" IMHO unless we have a definition of "respectable company" in Red Hat, that is a very subjective and fluid term that I have no solid understanding of. I know well known companies that have EDA systems building systems on top of the InMemoryChannel where event loss is possible. Why is that considered invalid?
* "such data forging" that sentence has a huge presumption on the implementation and an exaggeration of the intention of the suggestion. I always get worried when talking in extremes. Maybe the reason is a misunderstanding of the suggestion itself. Definitely the intention was "when applicable". So I'd really appreciate if before stating such strong sentences you'd point out anything I might've missed. I made it more more clear here (is in the updated suggestion abve):
```
verify events receipt either via safe validation events (heartbeat, ephemeral events, test events) if your use case allows it or via checking Kafka topic for committed offsets. After events receipt verification, it is safe to continue using your subscriptions as usual
```

Reply:
Of course, it depends. As I said, it depends on industry, what data is being processed, etc. 

My point on this is, we need to provide a solution for everybody. By adding, just "if possible", we are not telling what to do in "else ..." situation. And its tuns out that many important users, are from financial, or government sectors, and those sectors are usually highly regulated. **We need solution for them as well**.

And if we find a solution that can be used in regulated industries, I think the same solution we can advise to less regulated users as well. And that simplify the whole advise by removing the `if regulated() then ... else ...` structure.